test/e2e/serial/tests/configuration.go

```diff
@@ -693,9 +697,195 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+					return
+				}
+
+				Eventually(func(g Gomega) {
+					err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+					Expect(err).ToNot(HaveOccurred())
+
+					applyNewTopologyManagerPolicy(&kcObj.TopologyManagerPolicy, initialTopologyManagerPolicy)
+
+					err = fxt.Client.Update(context.TODO(), targetedKC)
+					g.Expect(err).ToNot(HaveOccurred())
+				}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+				By("waiting for mcp to start updating")
+				waitForMcpsCondition(fxt.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdating)
+
+				By("wait for mcp to get updated")
+				waitForMcpsCondition(fxt.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdated)
+			}()
+
+			var schedulerName string
+			nroSchedObj := &nropv1.NUMAResourcesScheduler{}
+			nroSchedKey := objects.NROSchedObjectKey()
+			err = fxt.Client.Get(context.TODO(), nroSchedKey, nroSchedObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroSchedKey.String())
+
+			schedDp, err := podlist.With(fxt.Client).DeploymentByOwnerReference(context.TODO(), nroSchedObj.GetUID())
+			Expect(err).ToNot(HaveOccurred(), "failed to get the scheduler deployment")
+
+			schedPods, err := podlist.With(fxt.Client).ByDeployment(context.TODO(), *schedDp)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(len(schedPods)).To(Equal(1))
+
+			schedulerName = nroSchedObj.Status.SchedulerName
+			Expect(schedulerName).ToNot(BeEmpty(), "cannot autodetect the TAS scheduler name from the cluster")
+
+			// restart the secondary-scheduler pod
+			By(fmt.Sprintf("deleting the NRO Scheduler object: %s", nroSchedObj.Name))
+			nrosched.CheckNROSchedulerAvailable(fxt.Client, nroSchedObj.Name)
+			err = fxt.Client.Delete(context.TODO(), &schedPods[0])
+			Expect(err).ToNot(HaveOccurred())
+			nrosched.CheckNROSchedulerAvailable(fxt.Client, nroSchedObj.Name)
+
+			By("modifying the topology manager policy under kubeletconfig to single-numa-node")
+			Eventually(func(g Gomega) {
+				err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(targetedKC), targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				applyNewTopologyManagerPolicy(&kcObj.TopologyManagerPolicy, v1beta1.SingleNumaNodeTopologyManagerPolicy)
+				err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = fxt.Client.Update(context.TODO(), targetedKC)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+			waitForMcpsCondition(fxt.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdated)
+
+			// getting the number of cpus that is required for a numa zone to make a TAE deployment
+			const (
+				NUMAZonesRequired     = 2
+				hostsRequired         = 2
```

Comment:
for this test one node is also enough

Reply:
np

Reply:
this check should be done at the beginning of the test code, to avoid as much as we can any unnecessary churn and keep the test execution time at bay